### ISSUE TITLE
Workers stop heartbeating

### ISSUE THREAD

**Issue by databus23** [concourse/concourse#750]
We are seeing workers disappearing for a couple of weeks now.
When running 2.2.1 we had multiple occurrences of all workers (darwin & linux) not reconnecting after an atc/tsa outage. They stopped heartbeating with a failure and never resumed heartbeating.

Now that we have upgraded to 2.4.0 workers start disappearing at random without tsa going down.
There are not even errors logged by the workers. They just stop heartbeating. Restarting the workers makes them reregister with tsa again.
The least log messages related to heartbeating are:
```
Nov 03 15:52:12 mo-a5b515ae6 concourse[30611]: {"timestamp":"1478188332.755291224","source":"tsa","message":"tsa.connection.forward-worker.heartbeat.start","log_level":1,"data":
Nov 03 15:52:12 mo-a5b515ae6 concourse[30611]: {"timestamp":"1478188332.758820534","source":"tsa","message":"tsa.connection.forward-worker.heartbeat.reached-worker","log_level":0,"data":{"remote":"10.46.14.60:50180","session":"8.1.202","took":"3.308641ms"}}
Nov 03 15:52:12 mo-a5b515ae6 concourse[30611]: {"timestamp":"1478188332.763972282","source":"tsa","message":"tsa.connection.forward-worker.heartbeat.done","log_level":1,"data":{"remote":"10.46.14.60:50180","session":"8.1.202","worker-address":"100.77.0.15:34107","worker-platform":"linux","worker-tags":""}}
```

netstat shows me that the workers still have a connection open to tsa:2222 but it seems like nothing is going over the wire anymore.
I'm not ruling out intermittent network problems in our infrastructure causing this but in my mind the workers should be able to recover from this in all cases as soon as the network is fine again.
Could it be that the heartbeating gets stuck and doesn't kill the tcp connection in all cases? I'm not familiar with how the heartbeating control loop works.

We are running concourse binaries. The web component runs in k8s and we have workers running on VMs outside of the k8s cluster.

**Comment by ahume** [comment]
@databus23 Can I ask why you choose not to run workers in the k8s cluster? I'm currently looking in to doing this and am wondering if this is the wrong path?

**Comment by databus23** [comment]
Just happened again for a darwin worker (2.4.0).
These are the last log messages, after that just silence:
```
{"timestamp":"1478356466.198215246","source":"tsa","message":"tsa.connection.forward-worker.heartbeat.reached-worker","log_level":0,"data":{"remote":"10.46.14.12:52139","session":"13.1.1853","took":"44.578279ms"}}
{"timestamp":"1478356466.203379154","source":"tsa","message":"tsa.connection.forward-worker.heartbeat.done","log_level":1,"data":{"remote":"10.46.14.12:52139","session":"13.1.1853","worker-address":"100.77.0.15:40473","worker-platform":"darwin","worker-tags":""}}
{"timestamp":"1478356499.971666336","source":"worker","message":"worker.beacon.keepalive.failed","log_level":2,"data":{"error":"EOF","session":"3.4"}}
{"timestamp":"1478356499.971748114","source":"worker","message":"worker.beacon.restarting","log_level":2,"data":{"error":"wait: remote command exited without exit status or exit signal","session":"3"}}
{"timestamp":"1478356515.026731014","source":"worker","message":"worker.beacon.restarting","log_level":2,"data":{"error":"failed to dial: failed to connect to TSA: dial tcp 10.46.2.199:2222: i/o timeout","session":"3"}}
{"timestamp":"1478356530.048924446","source":"worker","message":"worker.beacon.restarting","log_level":2,"data":{"error":"failed to dial: failed to connect to TSA: dial tcp 10.46.2.199:2222: i/o timeout","session":"3"}}
```

**Comment by databus23** [comment]
@ahume This is off-topic but I don't think its the wrong path. We just happen to have resource limits (disk space) on our k8s cluster that don't make it a good fit for concourse workers atm.

**Comment by databus23** [comment]
Ok this still happens to us on a regular basis and I tried to diagnose it a bit more:

We had two workers gone missing today. I stopped the remaining three workers that where still working and then did a netstat on the `concourse web` side:
```
root@web-1865638648-7xo88:/# netstat -ntp |grep 2222
tcp     1120      0 100.70.0.6:2222         10.46.14.60:51636       ESTABLISHED 7/concourse
tcp     1504      0 100.70.0.6:2222         10.46.14.60:58144       ESTABLISHED 7/concourse
```
Ok two connections remaining which I assume belong to the workers which are missing. Notice that both connections have bytes left in the `Recv-Q` buffer.

Then I send SIGQUIT to both workers killing them.
After that netstat on the `concourse web` side looked like this:
```
root@web-1865638648-7xo88:/# netstat -ntp |grep 2222
tcp     1120      0 100.70.0.6:2222         10.46.14.60:51636       ESTABLISHED 7/concourse
tcp     1505      0 100.70.0.6:2222         10.46.14.60:58144       CLOSE_WAIT  7/concourse
```
So one stuck in `CLOSE_WAIT` one still stuck in `ESTABLISHED` despite the other side being gone as the worker process were killed.

I finally send a SIGQUIT to the `concourse web` process.

All three stack dumps can be found here: https://mdocs.sap.com/mcm/public/v1/open?shr=9_XyxXHo4gs1SUgU05VCOtiUiS86azHDOpo17FGoX8E

Unfortunately I can't really tell which workers stackdump belongs to which hanging tcp connection.

**Comment by vito** [comment]
Guessing this is your two dropped workers:

![screenshot from 2016-11-15 16-50-59](https://cloud.githubusercontent.com/assets/1880/20330428/bd77e486-ab53-11e6-8b1c-8c2f04b82072.png)

FWIW, sigquit'ing only the `web` node would have been better (otherwise the connection will just break which may unwedge things we'd be looking for), but I think it didn't matter in this case.

I think our handling of SSH channels and requests has to be made asynchronous.

**Comment by macb** [comment]
Seems like some timeouts may be a good first step instead of moving to all async.

**Comment by vito** [comment]
@macb What I'm saying is our implementation of the SSH server is likely incorrect precisely because it's not asynchronous. It's not a connection timeout, it's a logic error.

**Comment by databus23** [comment]
@vito What tool are you using to visualise the stackdump?

Not sure if this is related but I also noticed that workers are going away when streaming inputs takes some time (longer then the heartbeat interval?). Seems to clog up the forwarded pipe. While streaming inputs I that `tsa.connection.forward-worker.heartbeat.reached-worker` messages suddenly come in with something like `took: "2m16s"` while normally its something like `"took":"3.451177ms"`.

Whats the purpose of the `beacon` loop in the worker? I thought it was to keep the connection alive. But what I see in our cases is that according to tcpdump that not a single packet is going over the wire anymore. Does this make sense? Shouldn't the connection break at some point and the worker try to reconnect? Does this mean the worker is stuck as well?

**Comment by databus23** [comment]
We just had another worker go away.
I took another stackdump and published under the link above. This time I didn't `SIGQUIT` the worker before.

This is what netstat told me before killing the web component:
```
root@web-1865638648-7xo88:/# netstat -ntp |grep 2222
tcp        0      0 100.70.0.6:2222         10.46.14.60:44420       ESTABLISHED 7/concourse
tcp     2112      0 100.70.0.6:2222         10.46.14.60:58242       ESTABLISHED 7/concourse
tcp        0      0 100.70.0.6:2222         10.46.14.60:49458       ESTABLISHED 7/concourse
tcp        0      0 100.70.0.6:2222         10.46.14.60:47608       ESTABLISHED 7/concourse
tcp        0      0 100.70.0.6:2222         10.46.14.12:65283       ESTABLISHED 7/concourse
```
So again one connection stuck where the web component is not reading from the socket.

Whats maybe interesting that restarting the web component (by SIGQUIT) caused all workers to go away. Not a single one reconnected.
Checking on each (working) worker they each still had an `ESTABLISHED` tcp connection to web:2222 while the remote process was long gone. 

They all just had long lines like this at the end:
```
{"timestamp":"1479298414.268207788","source":"worker","message":"worker.beacon.restarting","log_level":2,"data":{"error":"wait: remote command exited without exit status or exit signal","session":"3"}}
{"timestamp":"1479298415.324530602","source":"worker","message":"worker.beacon.keepalive.failed","log_level":2,"data":{"error":"EOF","session":"3.1"}}
```

**Comment by databus23** [comment]
Started to dig a little bit into the tsa code and found out that `github.com/felixge/tcpkeepalive` is used to set keepalives on the SSH connections.

As the [README](https://github.com/felixge/tcpkeepalive#tcpkeepalive) of the library hints that there are known issues with the library.
From: https://groups.google.com/forum/#!msg/golang-nuts/rRu6ibLNdeI/TIzShZCmbzwJ

> Don't use the keepalive library linked in that HOWTO article. It leaks file descriptors, and leaves each connection in blocking mode. 

I wonder if the blocking mode this library seems to enforce is responsible for this lockup.

**Comment by databus23** [comment]
Reading [this blog post](http://felixge.de/2014/08/26/tcp-keepalive-with-golang.html) of the author of `github.com/felixge/tcpkeepalive` made me wonder if using `SetReadDeadline` of the `net.Conn` interface on both ends of the connection would be more robust for detecting stale network connections.

As both ends send regular beacons/heartbeats through the connection a reasonable deadline on successive read calls should be a reliable indicator that the connection went stale, no?

**Comment by vito** [comment]
Agreed. I was looking through the stack dumps and notice that the beacon was hanging on the keepalive, but interestingly it was hanging in client-side locking, and I didn't see where the holder of the lock was as I got pulled away to something else. My concern is that we haven't yet identified whether it's a logic error or a network fault, but I do think that tcpkeepalives are something we should have regardless.

**Comment by databus23** [comment]
@vito Anything more I can provide to get to the bottom of this? I'm at a point where every time I restart the atc, which I have been doing a lot lately because of #785, all workers are not reconnecting automatically. I have to restart them manually.

**Comment by vito** [comment]
I'll start with wiring in `tcpkeepalive`. We already use it when talking to Garden. We use the non-blocking interface, so it should be a drop-in thing.

I'm still concerned that we're not properly handling things asynchronously in the TSA, but looking at the stack dumps I'm not convinced that's the problem, so I'll start with the simplest fix that you can test.

**Comment by vito** [comment]
@databus23 Give https://github.com/concourse/bin/releases/tag/v2.5.1-rc.20 a try - it's got `tcpkeepalive` wired in.

**Comment by databus23** [comment]
@vito. Thanks, I redeployed our workers with v.2.5.1-rc.20. I'll keep you posted.

**Comment by databus23** [comment]
@vito Are you not concerned about these [statements](https://github.com/concourse/concourse/issues/750#issuecomment-261673059) about github.com/felixge/tcpkeepalive?

**Comment by vito** [comment]
@databus23 no - they're referring to a different part of the library than
we use. we ran into that before and switched to the nonblocking variant. it
is a bit odd though to have such a subtle pitfall next to the safe
interface.

**Comment by vito** [comment]
@databus23 any improvement yet? or too early to tell?

**Comment by databus23** [comment]
@vito
I just lost another worker running v.2.5.1-rc.20. The worker logs just show that the heartbeating from tsa stopped 25min ago. The web component is running for 4 hours without interuptions.

netstat on the worker:
```
$> netstat -ntp |grep 2222
tcp        0      0 10.67.74.120:51552      10.46.2.199:2222        ESTABLISHED 61923/concours
```
 netstat on the web component:
```
$> netstat -ntp | grep 2222
tcp        0      0 100.65.0.27:2222        10.46.14.60:43342       ESTABLISHED 7/concourse
tcp        0      0 100.65.0.27:2222        10.46.14.60:47376       ESTABLISHED 7/concourse
tcp        0      0 100.65.0.27:2222        10.46.14.60:40094       ESTABLISHED 7/concourse
tcp        0      0 100.65.0.27:2222        10.46.14.60:55268       ESTABLISHED 7/concourse
tcp        0      0 100.65.0.27:2222        10.46.14.60:38246       ESTABLISHED 7/concourse
tcp     2400      0 100.65.0.27:2222        10.46.14.60:51552       ESTABLISHED 7/concourse
tcp        0      0 100.65.0.27:2222        100.65.0.1:49851        ESTABLISHED 7/concourse
```
*Note the next to last line, where the Q-receive receive buffer is not empty indicating TSA side is stuck.*

using tcpdump on the worker shows that keep alive packets are still happily flowing every 10 seconds:
```
tcpdump -pni ens160 -v "tcp port 2222 and ( tcp[tcpflags] & tcp-ack != 0 and ( (ip[2:2] - ((ip[0]&0xf)<<2) ) - ((tcp[12]&0xf0)>>2) ) == 0 ) "
tcpdump: listening on ens160, link-type EN10MB (Ethernet), capture size 262144 bytes
13:58:58.041641 IP (tos 0x0, ttl 64, id 63843, offset 0, flags [DF], proto TCP (6), length 52)
    10.67.74.120.51552 > 10.46.2.199.2222: Flags [.], cksum 0x61d6 (incorrect -> 0x3ea5), ack 1801661069, win 11045, options [nop,nop,TS val 198740852 ecr 19493060], length 0
13:58:58.042227 IP (tos 0x0, ttl 54, id 57980, offset 0, flags [DF], proto TCP (6), length 52)
    10.46.2.199.2222 > 10.67.74.120.51552: Flags [.], cksum 0x0e20 (correct), ack 1, win 4052, options [nop,nop,TS val 19503076 ecr 197701689], length 0
13:59:08.057629 IP (tos 0x0, ttl 64, id 63844, offset 0, flags [DF], proto TCP (6), length 52)
    10.67.74.120.51552 > 10.46.2.199.2222: Flags [.], cksum 0x61d6 (incorrect -> 0x0dbd), ack 1, win 11045, options [nop,nop,TS val 198743356 ecr 19503076], length 0
13:59:08.058158 IP (tos 0x0, ttl 54, id 57981, offset 0, flags [DF], proto TCP (6), length 52)
    10.46.2.199.2222 > 10.67.74.120.51552: Flags [.], cksum 0xe6ff (correct), ack 1, win 4052, options [nop,nop,TS val 19513092 ecr 197701689], length 0
```

So to me this seems like  a goroutine or something is stuck at the tsa side.

The stackdump from the web component con be found [here](https://mdocs.sap.com/mcm/public/v1/open?shr=9_XyxXHo4gs1SUgU05VCOtiUiS86azHDOpo17FGoX8E&obj=6BO02ScXgQdZhSeXgorFpouyoiM6xE5xaDIqvVbEPJ8)

**Comment by databus23** [comment]
After stoping the worker, the connection on the TSA side goes into `CLOSE_WAIT`
`tcp     2401      0 100.65.0.27:2222        10.46.14.60:51552       CLOSE_WAIT  7/concourse`

**Comment by databus23** [comment]
And I just noticed that since we lost the worker we have several job stuck in pending as described in #785 because of missing check containers.

I'm assuming thats because some of the check containers where located on that worker. So maybe at least for us the two issues are related.

**Comment by databus23** [comment]
Just happened again with a different worker. Exactly the same pattern

**Comment by vito** [comment]
OK - will switch to investigating the TSA side of things. I'll change it to process channels in parallel and let you know when there's a new RC.

**Comment by databus23** [comment]
@vito Thanks. I'm a right to assume that v2.5.1-rc.26 is the next one to test  ?

**Comment by vito** [comment]
@databus23 Yep. Here you go: https://github.com/concourse/bin/releases/tag/v2.5.1-rc.26

**Comment by beeva-carlosleon** [comment]
The new version 2.5.1 solve this issue?

Thanks!

**Comment by databus23** [comment]
So far it looks good. I have 2.5.1-rc.26 running on a fairly busy system for 3 days running without workers dropping off. I consider this solved when it keeps running for a week.

I'm still seeing the problem of workers not reconnecting when I restart the atc but I think this is a different problem I will open a separate issue for at some point. /cc @vito

**Comment by databus23** [comment]
Ok now it happened again 😭.

I was running the web component with the `concourse/concourse-rc:2.5.1-rc.26` docker image.

One worker dropped of after a little over 3 days and the symptoms are the same as before:
 * heartbeats from the tsa just stop.
 * keepalives from the worker are send and answered every 10 seconds.
 * two jobs a are hanging in pending, presumably because check containers are missing.
 * Killing the worker leaves the connection on the TSA side stuck in `CLOSE_WAIT`
 * Only difference: No bytes in the `Q-Recv` buffer on the TSA side for that connection.

[Stackdump from the web component](https://mdocs.sap.com/mcm/public/v1/open?shr=9_XyxXHo4gs1SUgU05VCOtiUiS86azHDOpo17FGoX8E&obj=EWUxQkPbieZmMi9EKfIDVgeR9M5p46YhYbuBIFRcmfE)

I update to 2.5.1 now and restart.

@vito I'm wondering if maybe processing the channels chan asynchronously is not enough. Maybe the requests chan also needs to be processed asynchronously.  Could it be that some operation in the `for req := range requests` loop blocks?

**Comment by vito** [comment]
We're starting to believe this is a deadlock internal to the `golang.org/x/crypto/ssh` package.

It happened to us with both our Darwin and Windows workers, so we dug in a bit. I found that upon bouncing the TSA (which I did to collect the stack dumps), the workers came back, so that tells me it's a TSA-side deadlock.

And the reason it takes some time to occur is there's a key renegotiation process that occurs after 1GB of data (by default) is sent over the connection. This is especially likely to happen for external workers that have data streamed over the connection (i.e. resources).

Here's our investigating:

* [`(*handshakeTransport).readOnePacket`](https://github.com/golang/crypto/blob/abc3c1b02083efdbef727c8631a865ad7d8d8b19/ssh/handshake.go#L151) and [`(*handshakeTransport).writePacket`](https://github.com/golang/crypto/blob/abc3c1b02083efdbef727c8631a865ad7d8d8b19/ssh/handshake.go#L311) check if the number of bytes exceed `t.config.RekeyThreshold`, which defaults to 1 gigabyte
* this will do some SSH protocol stuff and set `t.sentInitMsg` to a non-nil value (relevant later)
* [`(*handshakeTransport).writePacket` has a loop on `t.sentInitMsg != nil`](https://github.com/golang/crypto/blob/abc3c1b02083efdbef727c8631a865ad7d8d8b19/ssh/handshake.go#L314) using a condition variable (i.e. it's waiting for that to become `nil`, so the kex has finished, or for there to be a write error that will cause it to give up)
* `t.sentInitMsg` only becomes nil in a call to [`(*handshakeTransport).readOnePacket`](https://github.com/golang/crypto/blob/abc3c1b02083efdbef727c8631a865ad7d8d8b19/ssh/handshake.go#L191)
* the only caller of `readOnePacket` is [`(*handshakeTransport).readLoop`](https://github.com/golang/crypto/blob/abc3c1b02083efdbef727c8631a865ad7d8d8b19/ssh/handshake.go#L129)
* in our stack, we observed `readLoop` is blocked on [writing to the `incoming` channel](https://github.com/golang/crypto/blob/abc3c1b02083efdbef727c8631a865ad7d8d8b19/ssh/handshake.go#L138), so it cannot call `readOnePacket`
* who reads from `t.incoming`? [`(*handshakeTransport).readPacket`](https://github.com/golang/crypto/blob/abc3c1b02083efdbef727c8631a865ad7d8d8b19/ssh/handshake.go#L119) does.
* who calls `readPacket`? [`(*mux).onePacket`](https://github.com/golang/crypto/blob/abc3c1b02083efdbef727c8631a865ad7d8d8b19/ssh/mux.go#L215)
* who calls `onePacket`? `(*mux).loop` - which isn't too interesting though since it literally just keeps calling that method as long as it works. so let's go back into `onePacket` and see what it's doing *after* reading the packet, as that'd be why it's not reading the next one.
* in our case, it was in the call to [`(*channel).handlePacket`](https://github.com/golang/crypto/blob/abc3c1b02083efdbef727c8631a865ad7d8d8b19/ssh/mux.go#L246) - so what's *that* doing?
* in our case, it's blocked on a [call to `(*channel).sendMessage`](https://github.com/golang/crypto/blob/abc3c1b02083efdbef727c8631a865ad7d8d8b19/ssh/channel.go#L403). things are getting interesting here: our read loop is now writing.
* following the stack down, this ends up back in `(*handleTransport).writePacket`! which is dependent on us reading to succeed, as `readOnePacket` is what completes the key exchange.

**Comment by databus23** [comment]
@vito very impressive debugging. What do you mean by "external workers" though? Aren't all workers external? We are seeing this mostly with our linux workers which do a lot of stuff. They are using the forwarding/reverse tunnel thingy so I think in that case everything goes through the ssh connection.
I spend a little of time looking at the stack dumps (clearly not as much as you did) but I also got the feeling from where the goroutines are blocking that it might be inside the ssh package and not related to the TSA code. I'm just wondering why this is wasn't a problem in earlier versions of concourse. Seems to me we have this problem only in the past 3 month or so.
I see that the  `golang.org/x/crypto/ssh` dependency was last bumped 4 month ago. Maybe that brought in the deadlock behavior?

**Comment by vito** [comment]
@databus23 by external workers I mean workers doing the forwarding dance.
which is usually only necessary for those external to the private network,
but it's also the default for the binaries (so that they're secure out of
the box in more scenarios).

I haven't checked if the dependency bump is what triggered this, but I'd be
a little surprised. that sounds like a reasonable next step to take at
least.

armed with this new info about 1gb of data transfer we could also try
putting together a repro by lowering that number to force key exchanging to
happen more frequently.

also the smoking gun in the stack dump is a blocking "chan send" on the
line where it writes to the "incoming" channel.

**Comment by databus23** [comment]
@vito 
![image](https://cloud.githubusercontent.com/assets/439223/21394788/4d72afdc-c79a-11e6-9237-64a9e63e9299.png)

Looks like my linux workers suffer from the same bug.  Just had one worker drop off again.

**Comment by vito** [comment]
Opened https://github.com/golang/go/issues/18439

**Comment by databus23** [comment]
@vito: What's your take on the [suggested workaround](https://github.com/golang/go/issues/18439#issuecomment-271352326) to dramatically increase the re-key interval? Given that its still unclear how long it will take to fix this upstream I would appreciate this being implemented to improve the current situation. I'm also eager to start testing if we really got the root cause of all cases where workers go away unexpectedly.

**Comment by vito** [comment]
@databus23 I agree with your assessment in the other issue - while it would make this GH issue much less painful it seems a bit suspect from a security standpoint. I could only see that value being safe to increase if the strength of the crypto underlying SSH increased over 10 years, not the capable throughput of today's networks.

Here's the RFC that documents it: https://tools.ietf.org/html/rfc4344

**Comment by davidgarciaMontreal** [comment]
Is there a workaround for this? this same issue is happening in our production nodes.

The workers go stalled and then go into running states.

{"timestamp":"1507951457.896024466","source":"tsa","message":"tsa.connection.channel.forward-worker.heartbeat.failed-to-reach-worker","log_level":1,"data":{"baggageclaim-took":"1m0.001244755s","garden-took":"136.941764ms","remote":"142.133.130.64:39764","session":"8.1.1.9244"}}

 Containers: 1
  Running: 1
  Paused: 0
  Stopped: 0
 Images: 1
 Server Version: 17.09.0-ce
 Storage Driver: aufs
  Root Dir: /var/lib/docker/aufs
  Backing Filesystem: extfs
  Dirs: 10
  Dirperm1 Supported: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
 runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
 init version: 949e6fa
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 4.10.0-19-generic
 Operating System: Ubuntu 17.04
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 15.67GiB
 Name: rmriver-worker-05
 ID: U2FC:VNLF:JEGP:QJQK:4DT3:655V:Z35X:GHH6:WUIP:ZUJX:NVQH:ADKO
 Docker Root Dir: /var/lib/docker
 Debug Mode (client): false
 Debug Mode (server): false
 Registry: https://index.docker.io/v1/
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 
 WARNING: No swap limit support

root@a056c9367c9d:/# concourse --version
3.5.0

[291248.634968] device wr2jv39fe7i3-0 entered promiscuous mode
[291248.635389] wbrdg-0afe0064: port 1(wr2jv39fe7i3-0) entered blocking state
[291248.635395] wbrdg-0afe0064: port 1(wr2jv39fe7i3-0) entered forwarding state
[291248.635744] wbrdg-0afe0064: port 1(wr2jv39fe7i3-0) entered disabled state
[291248.743056] wbrdg-0afe0064: port 1(wr2jv39fe7i3-0) entered blocking state
[291248.743061] wbrdg-0afe0064: port 1(wr2jv39fe7i3-0) entered forwarding state
